subprojects/osgi-test/src/test/java/org/mockito/osgitest/OsgiTest.java

```diff
@@ -0,0 +1,122 @@
+import org.osgi.framework.launch.Framework;
+import org.osgi.framework.launch.FrameworkFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.fail;
+
+public class OsgiTest {
+
+    private static final FrameworkFactory frameworkFactory = ServiceLoader.load(FrameworkFactory.class).iterator().next();
+    private static final String STORAGE_TEMPDIR_NAME = "osgi-test-storage";
+    private static final List<String> EXTRA_SYSTEMPACKAGES = Arrays.asList("sun.misc", "sun.reflect");
+
+    private static final List<Path> TEST_RUNTIME_BUNDLES = splitPaths(System.getProperty("testRuntimeBundles"));
+    private static final String TEST_BUNDLE_SYMBOLIC_NAME = "testBundle";
+
+    private static final long STOP_TIMEOUT_MS = 10000;
+
+    private static Path frameworkStorage;
+    private static Framework framework;
+    private static Bundle testBundle;
+
+    @BeforeClass
+    public static void setUp() throws IOException, BundleException {
+        frameworkStorage = Files.createTempDirectory(STORAGE_TEMPDIR_NAME);
+        Map<String, String> configuration = new HashMap<>();
+        configuration.put(Constants.FRAMEWORK_STORAGE, frameworkStorage.toString());
+        configuration.put(Constants.FRAMEWORK_SYSTEMPACKAGES_EXTRA, String.join(",", EXTRA_SYSTEMPACKAGES));
+        framework = frameworkFactory.newFramework(configuration);
+        framework.init();
+        BundleContext bundleContext = framework.getBundleContext();
+        for (Path dependencyPath : TEST_RUNTIME_BUNDLES) {
+            Bundle installedBundle;
+            try {
+                installedBundle = bundleContext.installBundle(dependencyPath.toUri().toString());
+            } catch (BundleException e) {
+                throw new IllegalStateException("Failed to install bundle: " + dependencyPath.getFileName(), e);
+            }
+            if (TEST_BUNDLE_SYMBOLIC_NAME.equals(installedBundle.getSymbolicName())) {
+                testBundle = installedBundle;
+            }
+        }
+        if (testBundle == null) {
+            fail("Test bundle not found.");
+        }
+        framework.start();
+    }
+
+    @AfterClass
+    public static void tearDown() throws IOException, BundleException, InterruptedException {
+        try {
+            if (framework != null) {
+                framework.stop();
+                framework.waitForStop(STOP_TIMEOUT_MS);
+            }
+        } finally {
+            if (frameworkStorage != null) {
+                deleteRecursively(frameworkStorage);
+            }
+        }
+    }
+
+    @Test
+    public void testSimpleMock() throws BundleException, InterruptedException, ClassNotFoundException, IllegalAccessException, InstantiationException, IOException {
```

Comment:
Nit: let's make this `throws Exception` to be more concise.

Reply:
Makes sense. I collapsed multiple exception specifiers into `Exception` where applicable.